### DISCUSSION TITLE
How to generate sourcemaps for _next/static/chunks/* files?

### DISCUSSION THREAD

**Discussion by dtbaker** [vercel/next.js#17431] [Help]
Is there a way to generate sourcemaps for files under `/_next/static/chunks/*` ?

I'm receiving errors like this in rollbar:

```
Error: Failed to execute 'insertBefore' on 'Node': 
The node before which the new node is to be inserted is not a child of this node. 
at e (https://site/_next/static/chunks/framework.51cb2777f805d6a6985b.js:1:86576) 
at e (https://site/_next/static/chunks/framework.51cb2777f805d6a6985b.js:1:86649) 
at mo (https://site/_next/static/chunks/framework.51cb2777f805d6a6985b.js:1:86701) 
```

I have followed some instructions and have _most_ source maps uploading to rollbar correctly like so:

![image](https://user-images.githubusercontent.com/543073/94498268-13009d00-023d-11eb-8644-ab018a735e91.png)

However the ones under `_next/static/chunks/` do not seem to be generated or uploaded to rollbar (yellow in screenshot error in rollbar, unable to find matching sourcemap for reported error)?

I found this discussion which mentioned `static/chunks` but unsure if that's related to my issue here: https://github.com/vercel/next.js/discussions/7579

Cheers

**Comment by jamesmosier** [comment]
Take a look at this example from the [rollbar-sourcemap-webpack-plugin](https://github.com/thredup/rollbar-sourcemap-webpack-plugin/tree/master/examples/next-js) repo. I use that plugin with Next to upload sourcemaps to Rollbar with meaningful errors pointing to actual code from my source files.

**Reply by dtbaker** [reply]
Thanks @jamesmosier , That's the example I followed and am using it `withPlugins` like so:

```
const withPlugins = require('next-compose-plugins')

const RollbarSourcemapPlugin = require('rollbar-sourcemap-webpack-plugin')
const sourceMaps = require('@zeit/next-source-maps')

module.exports = withPlugins(
  [
    [sourceMaps, {}]
    .... other plugins here
  ],
...
```

Rollbar is reporting that sourcemaps are uploaded via API successfully like so:

![image](https://user-images.githubusercontent.com/543073/94499215-78ee2400-023f-11eb-9a8f-8d58484f1424.png)


However when certain errors come through, rollbar is unable to find the sourcemap. It seems to be with files under `static/chunks/*` 

![image](https://user-images.githubusercontent.com/543073/94499270-90c5a800-023f-11eb-8433-1dec7549a22f.png)

**Reply by jamesmosier** [reply]
Ah I see! Sorry about that, I misunderstood. I would think this would be an issue with the sourcemap upload plugin then 🤔  When a build occurs the sourcemaps for those `chunks/*` files get emitted, but I wonder if somehow that plugin isn't picking them up [here](https://github.com/thredup/rollbar-sourcemap-webpack-plugin/blob/master/src/RollbarSourceMapPlugin.js#L62-L91)

**Comment by digitalsanity** [comment]
Likely unrelated, but I had _next/ static chunk errors that I tracked down to needing to remove a final slash from my nginx proxy_pass.

**Comment by robbeman** [comment]
For future googlers, there is now a built-in option: https://nextjs.org/docs/pages/api-reference/next-config-js/productionBrowserSourceMaps

**Reply by ksaitor** [reply]
This would expose source maps to everyone, not only to rollbar. Would not recommend using this method in prod.
